src/phpDocumentor/ApiReference/Factory.php

```diff
@@ -0,0 +1,74 @@
+<?php
+/**
+ * phpDocumentor
+ *
+ * PHP Version 5.5
+ *
+ * @copyright 2010-2015 Mike van Riel / Naenius (http://www.naenius.com)
+ * @license   http://www.opensource.org/licenses/mit-license.php MIT
+ * @link      http://phpdoc.org
+ */
+
+namespace phpDocumentor\ApiReference;
+
+use phpDocumentor\DocumentGroup;
+use phpDocumentor\DocumentGroupDefinition as DocumentGroupDefinitionInterface;
+use phpDocumentor\DocumentGroupFactory;
+use phpDocumentor\Reflection\DocBlockFactory;
+use phpDocumentor\Reflection\Php\Factory\File\FlySystemAdapter;
+use phpDocumentor\Reflection\Php\NodesFactory;
+use phpDocumentor\Reflection\Php\ProjectFactory;
+use phpDocumentor\Reflection\Php\Factory as ProjectFactoryStrategy;
+use phpDocumentor\Reflection\PrettyPrinter;
+
+final class Factory implements DocumentGroupFactory
+{
+    /**
+     * Creates Document group using the provided definition.
+     *
+     * @param DocumentGroupDefinitionInterface $definition
+     * @return DocumentGroup
+     */
+    public function create(DocumentGroupDefinitionInterface $definition)
+    {
+        if ($definition instanceof DocumentGroupDefinition) {
+
+            $projectFactory = new ProjectFactory(
```

Comment:
this was the point where I needed your opinion. Like you can see on line 42 a new strategy is created using the current handled definition. Since the strategy needs its file system. 

But since I can only addl strategies via the constructor this is a problem. We could leave it like this or adapt the Project factory to add extra strategies. via a method. The latter means that we also need a remove strategy method. because the factory could be reused in a second iteration and the file system strategy is only needed for this specific definition...